HW14_1.py

```diff
@@ -0,0 +1,68 @@
+phone_book = {
+    'Igor': 132434,
+    'Diana': 254345,
+}
+
+import json
+import re
+
+json_data = json.dumps(phone_book)
+with open (r'C:\01_Education\03_IT\04_Pyhon\Laba\HW\phone_book.txt', 'a') as file:
+    file.write(json_data)
+
+print (phone_book.keys(),phone_book.values())
+
+while True:
+    user_input = input('Enter a command: ')
+    split_input = user_input.split()
+   
+    command = split_input[0]
+    
+    if len(split_input) > 1:
+        name = split_input[1]
+    else:
+        name = None
+
+    if len(split_input) > 2:
+        number = split_input[2]
+    else:
+        number = None      
+    
+    if command == 'stats':
+        print(len(phone_book))
+
+    elif command == 'add':
+        check_phone = re.findall(r'\+38\d+|38\d+|0', number)
```

Comment:
This pattern is bad for numbers in format 0631234567. add `\d+` after 0 to make it works. Also, you can additionally check for a number of digits using `d{10}` instead of `d+` (for 0 if will be d{9})
And since Ukrainian international code is 380 you can update your pattern to avoid false validation of numbers that starts with 381, for example

Reply:
Hello! Fixed. Thank you!

Reply:
It is still not ideal and can validate invalid formats. To improve it:
1. Use `re.match` instead of `re.findall`
2. Change number of digits: there should be 9 digits after `+38`, `38` or `0`, not 10
3. add `$` to every option to mark the end of the string and exclude such cases as `+380981234567a`

Reply:
It seems like fixed.